File: 62. Unique Paths/solution.java
class Solution {
    public int uniquePaths(int m, int n) {
        int[][] grid = new int[m][n];
        for (int i = 0; i < m; i++) {
            for (int j = 0; j < n; j++) {
                grid[i][j] = (i * j == 0) ? 1 : grid[i - 1][j] + grid[i][j - 1];
            }
        }
        return grid[m - 1][n - 1];
    }
}

File: 62. Unique Paths/solution.cpp
class Solution {
public:
    int uniquePaths(int m, int n) {
        unsigned long ans = 1;
        int small = min(m - 1, n - 1);
        for (int i = 1; i <= small; i++) {
            ans *= m + n - 1 - i;
            ans /= i;
        }
        return ans;
    }
};